packages/update-monitor/src/common/address.ts

```diff
@@ -1,5 +1,7 @@
-import { utils } from 'ethers'
+import { constants, utils } from 'ethers'
 
 export function bytes32ToAddress(bytes32: string) {
-  return utils.getAddress('0x' + bytes32.slice(-40))
+  return bytes32 === '0x'
+    ? constants.AddressZero
```

Comment:
I think that this will lead to problems down the line. If you call an EOA then `0x` will always be a return value which will silently return `0x00000...` as a valid address instead of throwing and alerting us.

Reply:
Makes sense, reverted. I moved the check to the specific function that needs it.